mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -154,9 +154,9 @@ private[spark] class MatrixUDT extends UserDefinedType[Matrix] {
         row.setByte(0, 0)
         row.setInt(1, sm.numRows)
         row.setInt(2, sm.numCols)
-        row.update(3, sm.colPtrs.toSeq)
-        row.update(4, sm.rowIndices.toSeq)
-        row.update(5, sm.values.toSeq)
+        row.update(3, new GenericArrayData(sm.colPtrs.map(_.asInstanceOf[Any])))
```

Comment:
this would be a lot more efficient as a loop.

maybe we should have a factory method that accepts various specialized arrays in GenericArrayData